pvanalytics/quality/irradiance.py

```diff
@@ -390,3 +391,66 @@ def clearsky_limits(measured, clearsky, csi_max=1.1):
         max_clearsky_index=np.Inf
     )
     return util.check_limits(csi, upper_bound=csi_max, inclusive_upper=True)
+
+
+def _to_hours(freqstr):
+    if freqstr[0].isalpha():
+        freqstr = '1' + freqstr
+    return pd.to_timedelta(freqstr).seconds / 3600
+
+
+def _daily_total(series):
```

Comment:
Thinking this could be generalized in case someone would rather use a trapezoid rule. Maybe resample in the calling function and this function does the integration?

Reply:
My preference would be a kwarg specifying the integration method. We can use `scipy.integration` to support the trapezoid rule and Simpson's rule as well as the left-hand rule.

Reply:
Worth trying `scipy.integration`, but it can be really slow

Reply:
One more option comes to mind. Have users pass a Series of daily insolation and provide functions in `pvanalytics.metrics` to calculate insolation either the way it is done here or with `scipy.integration`. This might give the best flexibility w.r.t. how insolation is calculated, but comes at the cost of only returning a daily mask, leaving the task of reindexing and applying the mask to the original data to the user.

Reply:
Came across this today: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.reindex_like.html may be useful here

Reply:
I'd suggest a trapezoid integration as the default way to get insolation from irradiance. For a regular spaced time series, it's `freq_hours * 0.5 * (series[:-1].sum() + series[1:].sum())` if that's correct code, and let's move on.

Reply:
Should we raise an error if the series does not have equal timestamp spacing (like `pvlib.clearsky.detect_clearsky`)?

Reply:
yes, unless we want to extract the timedeltas and use them in the integral

Reply:
There may be a trade off between flexibility and performance. I would prefer to keep the flexibility of uneven timestamps, but I suspect that there could be large performance penalties for extracting the time-deltas on each day. I'll need to do a little testing to see if I'm right, or how much it matters.

Reply:
`scipy.integrate.trapz()` seems to perform very well (faster than the snippet above). I added a basic test for uniform spacing, and use the scipy function to integrate. If the timestamp spacing is uneven then the time in hours since midnight is passed to `trapz` for each value otherwise the frequency in hours is used as dx. The performance gain for having even timestamp spacing is ~4x on 8 years of 15 minute irradiance data, in both cases I think the performance is adequate.

Reply:
84217d7 makes the performance for uneven timestamps slightly worse than what I described above; however, I am still happy since the performance of the even timestamp case is unaffected.